Exercise

Use iris flower dataset from sklearn library and use cross_val_score against following models to measure the performance of each. In the end figure out the model with best performance,

Logistic Regression

SVM

Decision Tree

Random Forest

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.svm import SVC
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import KFold
from sklearn.model_selection import train_test_split
from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import cross_val_score
from sklearn.tree import DecisionTreeClassifier

In [2]:
iris = load_iris()

### Logistic Regression

In [3]:
log_reg_cross_val_s = cross_val_score(LogisticRegression(max_iter=1000), iris.data, iris.target)
print(f'Result of Logistic Regression: {log_reg_cross_val_s}')
print(f'Average Result of Logistic Regression: {np.average(log_reg_cross_val_s)}')

Result of Logistic Regression: [0.96666667 1.         0.93333333 0.96666667 1.        ]
Average Result of Logistic Regression: 0.9733333333333334


### SVM

In [4]:
svm_cross_val_s = cross_val_score(SVC(), iris.data, iris.target)
print(f'Result of Support Vector Machine: {svm_cross_val_s}')
print(f'Average Result of Support Vector Machine: {np.average(svm_cross_val_s)}')

Result of Support Vector Machine: [0.96666667 0.96666667 0.96666667 0.93333333 1.        ]
Average Result of Support Vector Machine: 0.9666666666666666


### Decision Tree

In [8]:
dec_tree_cross_val_s = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
print(f'Result of Decision Tree Classifier: {dec_tree_cross_val_s}')
print(f'Average Result of Decision Tree Classifier: {np.average(dec_tree_cross_val_s)}')

Result of Decision Tree Classifier: [0.96666667 0.96666667 0.9        1.         1.        ]
Average Result of Decision Tree Classifier: 0.9666666666666668


### Random Forest Classifier

In [7]:
random_forest_cross_val_s = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target)
print(f'Result of Random Forest Classifierr: {random_forest_cross_val_s}')
print(f'Average Result of Random Forest Classifier: {np.average(random_forest_cross_val_s)}')

Result of Random Forest Classifierr: [0.96666667 0.96666667 0.93333333 0.96666667 1.        ]
Average Result of Random Forest Classifier: 0.9666666666666668


#### Best score so far is from Logistic Regression: 0.9733333333333334